### PR TITLE
Minor game fixes for Arcade

### PR DESCRIPTION
**Pull request by jwunderl**
* only set _life to null if the value is not modified in the _lifeOverHandler event
    * currently, resetting life to some positive value in the _lifeOverHandler doesn't persist to the rest of the game
    * for example, you cannot choose to give just a single life to the player when it resets

* add in the Sprite.data field we keep forgetting to add, for maintaining the state of a given sprite (javascript only). Example usages:
    * keep the name a player chose for each sprite (naming animals in a farm game)
    * give bosses 'health'
    * store "items" / keys on enemies for an RPG

### CODE REVIEW

File: libs/game/sprite.ts [hunk 1]
```diff
@@ -63,6 +63,9 @@ class Sprite implements SpriteLike {
 
     _type: number;
 
+    // for tracking state of sprite; unused internally
+    data: string;
```
[the remark this answers is not in the document]
Comment (jwunderl) [reply libs/game/sprite.ts]: For the course, I'd probably only end up using it for the first two ideas I listed (names and health for enemies), and type conversion between numbers and strings seems like a reasonably useful topic to bring up in the course.

On the other hand, using numbers in Sprite.data to associate to names in a string[] would also be good for a lesson, though, so either way seems reasonable to me. I'll change it if it seems better to you.

**Comment by pelikhan** [comment]
@mmoskal should we have "data" by "any" and use interface to store rich data?

**Comment by mmoskal** [comment]
Once the compiler is updated (ETA 2 weeks or so) we can use `any`. But for now I think it's fine to use `string` or `number`

**Comment by jwunderl** [comment]
It's only two weeks out (ish) till we get any? Nice.

I definitely agree changing from string|number -> any would be ideal in the future (and wouldn't break any existing code).

**Comment by pelikhan** [comment]
Is this still relevant? Can you split the fixes?

**Comment by jwunderl** [comment]
was waiting on the ``any`` support to change sprite.data to type ``any``, I'll close this and do the info fix in another PR and wait on the Sprite.data till ``any`` is in

**Pull request closed without merging**